### PR TITLE
Created an Esdl asset mapper for ATES.

### PR DESCRIPTION
**Pull request by AshrafAlmohagryDeltares**

### CODE REVIEW

File: src/omotes_simulator_core/adapter/transforms/esdl_asset_mappers/EsdlAssetAtesMapper.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+from omotes_simulator_core.entities.assets.ates_cluster import AtesCluster
+from omotes_simulator_core.entities.assets.asset_abstract import AssetAbstract
+from omotes_simulator_core.entities.assets.esdl_asset_object import EsdlAssetObject
+from omotes_simulator_core.simulation.mappers.mappers import EsdlMapperAbstract
+
+
+class EsdlAssetAtesMapper(EsdlMapperAbstract):
+    """Class to map an ESDL asset to a ates entity class."""
+
+    def to_esdl(self, entity: AtesCluster) -> EsdlAssetObject:
```
Comment (vanmeerkerk) [inline src/omotes_simulator_core/adapter/transforms/esdl_asset_mappers/EsdlAssetAtesMapper.py:26]: The ATES cluster needs to have all the add_physical_data parts in it from the ATES asset.

File: src/omotes_simulator_core/adapter/transforms/esdl_asset_mapper.py [hunk 2]
```diff
@@ -60,11 +65,24 @@ def to_entity(model: EsdlAssetObject) -> AssetAbstract:
 
         :return: Entity object of type AssetAbstract.
         """
-        if not type(model.esdl_asset) in CONVERSION_DICT:
+        if (
```
Comment (vanmeerkerk) [inline src/omotes_simulator_core/adapter/transforms/esdl_asset_mapper.py:68]: First resolve https://github.com/Project-OMOTES/simulator-core/pull/194

File: src/omotes_simulator_core/adapter/transforms/mappers.py [hunk 3]
```diff
@@ -1,19 +1,5 @@
-#  Copyright (c) 2023. Deltares & TNO
-#
-#  This program is free software: you can redistribute it and/or modify
-#  it under the terms of the GNU General Public License as published by
-#  the Free Software Foundation, either version 3 of the License, or
```
Comment (vanmeerkerk) [inline src/omotes_simulator_core/adapter/transforms/mappers.py:5]: Why did you remove the copyright statement?

**Review by vanmeerkerk — CHANGES_REQUESTED** [review]
This mapper requires some changes to implement the add_physical_data, right?

File: src/omotes_simulator_core/adapter/transforms/mappers.py (hunk 3 above)
Comment (AshrafAlmohagryDeltares) [reply src/omotes_simulator_core/adapter/transforms/mappers.py]: it was by mistake, i put it back.

File: src/omotes_simulator_core/adapter/transforms/esdl_asset_mappers/EsdlAssetAtesMapper.py (hunk 1 above)
Comment (AshrafAlmohagryDeltares) [reply src/omotes_simulator_core/adapter/transforms/esdl_asset_mappers/EsdlAssetAtesMapper.py]: completed.

**Comment by AshrafAlmohagryDeltares** [comment]
I noticed a mistake in the add_physical_data in the ates_cluster (before removing it and adding it to the mapper). Both maximum_flow_charge and maximum_flow_discharge uses the same input from the esdl file "maxChargeRate" and still passes the test because the values in the esdl test file are the same (test_ates.esdl).

**Comment by MichielTukker** [comment]
@lfse-slafleur Test is failing because of a segfault in the Pyjnius/Ates code.  Disabling faulthandler causes a segfault on linux, but it's required on windows to prevent an access violation.  Any ideas?

**Comment by lfse-slafleur** [comment]
> @lfse-slafleur Test is failing because of a segfault in the Pyjnius/Ates code. Disabling faulthandler causes a segfault on linux, but it's required on windows to prevent an access violation. Any ideas?

Offline checked with Michiel that weirdly this issue isn't present on main. Going to investigate if the diff with main provides any clues.

**Review by MichielTukker — APPROVED** [review]

**Comment by lfse-slafleur** [comment]
> > @lfse-slafleur Test is failing because of a segfault in the Pyjnius/Ates code. Disabling faulthandler causes a segfault on linux, but it's required on windows to prevent an access violation. Any ideas?
> 
> Offline checked with Michiel that weirdly this issue isn't present on main. Going to investigate if the diff with main provides any clues.

Faulthandler lines in the code did not work in linux. We disabled it through the CLI flag `-p no:faulthandler` which appears to work for linux while removing the access violation messages on Windows.
